open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/metadatahighway/OMRSCohortManager.java

```diff
@@ -275,12 +302,71 @@ public void initialize(String                           cohortName,
                                   auditCode.getSystemAction(),
                                   auditCode.getUserAction(),
                                   error);
+            
+            return new CohortInitializationResult(error);
         }
 
         log.debug(actionDescription + " COMPLETE");
+        return CohortInitializationResult.SUCCESS;
     }
 
+    /**
+     * Updates the status of the connection.  
+     * 
+     * Package private so it can be called by {@link OMRSCohortManagerInitializer}
+     */
+    synchronized void setStatus(CohortConnectionStatus status) {
+        this.cohortConnectionStatus = status;
+    }
+    
+    private void unregisterListeners() {
 
+        unregisterCohortEventListener(cohortTopicConnector, cohortEventListener);
+        
+        if (cohortRepositoryEventManager != null) 
+        {
+            if (repositoryEventPublisher != null) 
+            {
+                cohortRepositoryEventManager.unregisterRepositoryEventProcessor(repositoryEventPublisher);
+            }
+            
+            if (enterpriseEventPublisher != null)
+            {
+             
+                cohortRepositoryEventManager.unregisterRepositoryEventProcessor(enterpriseEventPublisher);
+            }
+            
+            if (localRepository != null) 
+            {
+                if (localRepository.getIncomingTypeDefEventProcessor() != null)
+                {
+                    cohortRepositoryEventManager.unregisterTypeDefProcessor(localRepository.getIncomingTypeDefEventProcessor());
+                }
+                
+                if (localRepository.getIncomingInstanceEventProcessor() != null)
+                {
+                    cohortRepositoryEventManager.unregisterInstanceProcessor(localRepository.getIncomingInstanceEventProcessor());
+                }
+            }
+        }
+        
+        if (localRepositoryEventManager != null && repositoryEventPublisher != null)
+        {
+        
+            localRepositoryEventManager.unregisterRepositoryEventProcessor(repositoryEventPublisher);
+        }
+    }
+
+    private static void unregisterCohortEventListener(
```

Comment:
@mandy-chessell This method was static as an optimization to avoid to caller needing to pass an implicit *this*  reference that would be unused.  This allows the compiler to do additional optimizations.  However, it most likely little difference in practice, so I'm fine with removing it in this case.  In general I try to make methods static whenever they can be since very little bit helps.